NotebookExamples/csharp/Samples/HousingML.ipynb

```diff
@@ -5,7 +5,7 @@
       "source": [
         "#r \"nuget:Microsoft.ML,1.4.0-preview\"\n",
         "#r \"nuget:Microsoft.ML.AutoML,0.16.0-preview\"\n",
-        "#r \"nuget:Microsoft.Data.DataFrame,0.1.1-e190920-1\""
+        "#r \"nuget:Microsoft.Data.DataFrame,1.0.0-e190910-1\""
```

Comment:
Unfortunately we messed up some versions on our myget feed. The `1.0.0-` versions are older than the `0.1.1-` versions. If you are looking at the feed, note the date it was published.

The good news is that we have renamed the package to `Microsoft.Data.Analysis` and so all the versioning issues went away.

I can send a separate update to fix this to the latest package name and version, and fixing any code that needs updating.

Reply:
Please revert the notebook changes.

Reply:
@jonsequitur, I had already reverted them on my machine.  Eric, updated the version, and have updated it to the newer version.

The tests were failing because there was no way to specify the custom feed, with the new location for the project file.  I have added support for #r "nuget:restoresources=http://blah".

Unlike the F# repo, this needs to be specified before the #r nugets that rely on it.  So … first I suppose.

I am going to spend some time seeing if I can wire up the F# #r nuget processor, wich would enable me to rip out some of the code.

@eerhardt --- the primary benefit of reading the assets.json file, will be that there is less guesswork about where to find the file is, and so it should perform better in the resolution stage.  My main gripe, is there is no decent way to find the nuget root directory.


I will try to incorporate asset.json probing in the F# probing algorithm and thus the C# probing if I can successfully incorporate it.  However, using #I to specified probing roots for assemblies, has always been a thing in F# and we will want to continue with that including native probing, it's a nice developer scenario.

Reply:
> My main gripe, is there is no decent way to find the nuget root directory.

I believe that is what the [`FallbackPackagePathResolver`](https://github.com/NuGet/NuGet.Client/blob/69a673582c62a63e917d152d871617ac5031dee1/src/NuGet.Core/NuGet.Packaging/FallbackPackagePathResolver.cs#L15) in NuGet is for. It will locate where a NuGet package is for a given context.

However, doesn't the Interactive Kernel control where the nuget root directory is?

Reply:
It doesn't.

Reply:
Per our discussions yesterday, I'd prefer to merge this PR and then revisit the approach, since there a number of different bugs being fixed here, not just the native dependency resolution.